lib/internal/child_process.js

```diff
@@ -352,10 +352,10 @@ function onErrorNT(self, err) {
 ChildProcess.prototype.kill = function(sig) {
   var signal;
 
-  if (sig === 0) {
-    signal = 0;
-  } else if (!sig) {
+  if (!sig) {
```

Comment:
This condition has to be after `typeof sig === 'number'` otherwise signal `0` will be incorrectly converted to `SIGTERM`